digest/src/core_api/rt_variable.rs

```diff
@@ -145,6 +156,54 @@ where
     }
 }
 
+impl<T> SerializableState for RtVariableCoreWrapper<T>
+where
+    T: VariableOutputCore + UpdateCore + SerializableState,
+    T::BlockSize: IsLess<U256>,
+    Le<T::BlockSize, U256>: NonZero,
+    T::SerializedStateSize: Add<U1>,
+    <T::SerializedStateSize as Add<U1>>::Output: Add<T::BlockSize> + ArrayLength<u8>,
+    <<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output: Add<U1> + ArrayLength<u8>,
+    <<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output: ArrayLength<u8> + Sub<T::SerializedStateSize>,
+    <<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output: Sub<U1> + ArrayLength<u8>,
+    <<<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output as Sub<U1>>::Output: Sub<T::BlockSize> + ArrayLength<u8>,
+    <<<<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output as Sub<U1>>::Output as Sub<T::BlockSize>>::Output: ArrayLength<u8>,
+{
+    type SerializedStateSize = Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>;
```

Comment:
You can add ignore attribute for this associated type to work around the clippy warnings.

Reply:
I have checked that the current version of `clippy` doesn't treat this as a warning anymore. The first version that does it is `1.59.0`.
Does it make sense to update the version of clippy (to the current stable `1.62.0` or minimal version `1.59.0`)? Or is it better to add an attribute?

Reply:
Yes, you can update clippy's version in CI.